website/client/src/store/getters/tasks.js

```diff
@@ -34,12 +34,66 @@ function getTaskColor (task) {
   return 'best';
 }
 
-export function canDelete () {
-  return task => {
-    const isUserChallenge = Boolean(task.userId);
-    const activeChallenge = isUserChallenge
-      && task.challenge && task.challenge.id && !task.challenge.broken;
-    return !activeChallenge;
+export function canDelete (store) {
+  return (task, taskCategory, isTaskFromUserDashboard, group, challenge) => {
+    let isUserCanDeleteTask = isTaskFromUserDashboard;
+    const user = store.state.user.data;
+
+    switch (taskCategory) {
+      case 'challenge':
+        if (challenge && !isTaskFromUserDashboard) {
+          const isUserChallenge = user.id === challenge.leader.id;
+          isUserCanDeleteTask = isUserChallenge || task.challenge.broken;
+        } else {
+          isUserCanDeleteTask = task.challenge.broken;
+        }
+        break;
+      case 'group':
+        if (group && !isTaskFromUserDashboard) {
+          isUserCanDeleteTask = (group.leader
+            && group.leader._id === user._id) || (group.managers
+            && Boolean(group.managers[user._id]));
+        } else {
+          isUserCanDeleteTask = false;
+        }
+        break;
+      default:
+        break;
+    }
+    return Boolean(isUserCanDeleteTask);
+  };
+}
+
+export function canEdit (store) {
+  return (task, taskCategory, isTaskFromUserDashboard, group, challenge) => {
+    let isUserCanEditTask = isTaskFromUserDashboard;
+    const user = store.state.user.data;
+    switch (taskCategory) {
+      case 'challenge':
+        if (!isTaskFromUserDashboard) {
+          if (challenge) {
+            const isUserChallenge = user.id === challenge.leader.id;
+            isUserCanEditTask = isUserChallenge || task.challenge.broken;
```

Comment:
same here, even if the challenge is broken I think only the leader should be able to edit it on the challenge's task board. Also because the `broken` attribute should only exist on the user's copy of the task